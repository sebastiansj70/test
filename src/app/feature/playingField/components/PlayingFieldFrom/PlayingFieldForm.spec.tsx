import '@testing-library/jest-dom';
import { RenderResult, fireEvent, render, wait } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { PlayingFieldForm } from '.';
import React from 'react';
import { setTextEvent } from 'app/shared/utils/test';

describe('PlayingFieldForm', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof PlayingFieldForm> & {
    onSubmit: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      onSubmit: stub(),
      formTitle: 'test playing field from',
      handleListTicket: jest.fn(),
    };
    componentWrapper = render(<PlayingFieldForm {...componentProps} />);
  });

  it('shoult match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });

  it('debe fallar al enviar todos los campos que faltan', async () => {
    const elem = componentWrapper.container;
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });

    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(2);
    expect(spans[0].textContent).toBe('El campo id cancha es requerido.');
    expect(spans[1].textContent).toBe('El campo status Cancha es requerido.');
  });

  it('debe fallar al enviar un campo faltantes', async () => {
    const elem = componentWrapper.container;
    const idCancha = elem.querySelector('input[name="idCancha"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
      idCancha && fireEvent.change(idCancha, setTextEvent('idCancha', '1'));
    });

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });

    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(1);
    expect(spans[0].textContent).toBe('El campo status Cancha es requerido.');
  });

  it('deberia enviar', async () => {
    const elem = componentWrapper.container;
    const idCancha = elem.querySelector('input[name="idCancha"]');
    const statusCancha = elem.querySelector('input[name="statusCancha"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
      idCancha && fireEvent.change(idCancha, setTextEvent('idCancha', '1'));
    });

    await wait(() => {
      statusCancha &&
        fireEvent.change(statusCancha, setTextEvent('statusCancha', 'bueno'));
    });

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });

    const formSubmitted = componentProps.onSubmit.firstCall.args[0];

    expect(formSubmitted.idCancha).toBe(1);
    expect(formSubmitted.statusCancha).toBe('bueno');
  });
});
