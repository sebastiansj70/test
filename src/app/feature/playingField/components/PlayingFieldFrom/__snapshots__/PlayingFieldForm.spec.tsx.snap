// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayingFieldForm shoult match snapshot 1`] = `
<div>
  <form
    class="sc-AxjAm cieRug"
  >
    <h1
      class="sc-AxhUy ftqyIz"
    >
      test playing field from
    </h1>
    <input
      class="sc-AxhCb gLMSgW"
      name="idCancha"
      placeholder="NÃºmero de cancha"
      type="number"
    />
    <input
      class="sc-AxhCb gLMSgW"
      name="statusCancha"
      placeholder="Estado de la cancha"
    />
    <button
      class="sc-AxiKw hNdhxp"
      type="submit"
    >
      Registrar
    </button>
  </form>
</div>
`;
